src/main/kotlin/org/jetbrains/research/testspark/tools/llm/generation/PromptManager.kt

```diff
@@ -124,32 +125,48 @@ class PromptManager(
                     }
 
                     CodeType.LINE -> {
+                        // two possible cases: the line inside a method/function or inside a class
                         val lineNumber = codeType.objectIndex
-                        val psiMethod = getPsiMethod(cut, getMethodDescriptor(cut, lineNumber))!!
-
                         // get code of line under test
-                        val document = psiHelper.getDocumentFromPsiFile()
-                        val lineStartOffset = document!!.getLineStartOffset(lineNumber - 1)
-                        val lineEndOffset = document.getLineEndOffset(lineNumber - 1)
-
-                        val lineUnderTest = document.getText(TextRange.create(lineStartOffset, lineEndOffset))
-                        val method = createMethodRepresentation(psiMethod)!!
-                        val interestingClassesFromMethod =
-                            psiHelper.getInterestingPsiClassesWithQualifiedNames(cut, psiMethod)
-                                .map(this::createClassRepresentation)
-                                .toList()
-
-                        promptGenerator.generatePromptForLine(
-                            lineUnderTest,
-                            method,
-                            interestingClassesFromMethod,
-                            testSamplesCode,
-                        )
+                        val lineUnderTest = psiHelper.getDocumentFromPsiFile()!!.let { document ->
+                            val lineStartOffset = document.getLineStartOffset(lineNumber - 1)
+                            val lineEndOffset = document.getLineEndOffset(lineNumber - 1)
+                            document.getText(TextRange.create(lineStartOffset, lineEndOffset))
+                        }
+
+                        val psiMethod = getMethodDescriptor(cut, lineNumber)?.let { descriptor ->
+                            getPsiMethod(cut, descriptor)
+                        }
+                        /**
+                         * if psiMethod exists, then use it as a context for a line,
+                         * otherwise use the cut as a context
+                         */
+                        if (psiMethod != null) {
+                            val method = createMethodRepresentation(psiMethod)!!
+                            val interestingClassesFromMethod =
+                                psiHelper.getInterestingPsiClassesWithQualifiedNames(cut, psiMethod)
+                                    .map(this::createClassRepresentation)
+                                    .toList()
+
+                            return@Computable promptGenerator.generatePromptForLine(
```

Comment:
You can get the package name from psiHelper, to provide more information to prompt generator(as it was done for `CodeType.METHOD`

Reply:
Supplied the package.